app/controllers/events_controller.rb

```diff
@@ -0,0 +1,107 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :user_login, except: %i[index show]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+    @users = User.all
+    @today = []
+    @past = []
+    @upcoming = []
+    @events.each do |event|
+      if event.date == Date.today
+        @today.push(event)
+      elsif event.date > Date.today
+        @upcoming.push(event)
+      else
+        @past.push(event)
+      end
+    end
+  end
```

Comment:
###### `index` action

* [ ] lists past events, i.e. `@past_events = Event.previous_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists ucomping events, i.e. `@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

> Use the model scopes instead of getting the upcoming and past events with a loop